lms/views/api/canvas/authorize.py

```diff
@@ -100,18 +101,13 @@ def oauth2_redirect(request):
     renderer="lms:templates/api/canvas/oauth2_redirect_error.html.jinja2",
 )
 def oauth2_redirect_error(request):
-    template_variables = {
-        "invalid_scope": request.params.get("error") == "invalid_scope",
-        "details": request.params.get("error_description"),
-        "scopes": FILES_SCOPES + SECTIONS_SCOPES,
-    }
-
-    if request.lti_user:
-        authorization_param = BearerTokenSchema(request).authorization_param(
-            request.lti_user
-        )
-        template_variables["authorize_url"] = request.route_url(
-            "canvas_api.authorize", _query=[("authorization", authorization_param)],
-        )
-
-    return template_variables
+    # In an exception view the `context` is usually the exception that occurred.
+    # We're going to cheat here and create a different object instead to serve
+    # as the `context` in templates.
+    context = FrontendAppResource(request)
+    context.js_config.enable_oauth_error_mode(
+        error_details=request.params.get("error_description"),
+        is_scope_invalid=request.params.get("error") == "invalid_scope",
+        requested_scopes=FILES_SCOPES + SECTIONS_SCOPES,
+    )
+    return {"context": context}
```

Comment:
Since you're returning `return {"context": context}` to the template, I'm guessing some template somewhere does `context.js_config` but, if you don't `return {"context": context}`, then `context` is the exception not the `FrontendAppResource`?

If so, can the template just do `request.context.js_config` instead?